odoo/models.py

```diff
@@ -3671,12 +3686,10 @@ def _check_company(self, fnames=None):
             # with the company of the origin document, i.e. `self.account_id.company_id == self.company_id`
             for name in regular_fields:
                 corecord = record.sudo()[name]
-                # Special case with `res.users` since an user can belong to multiple companies.
-                if corecord._name == 'res.users' and corecord.company_ids:
-                    if not (company <= corecord.company_ids):
+                if corecord and ('company_id' in corecord._fields or corecord._name == 'res.company'):
```

Comment:
I don't understand the second part of the `and`. The condition `'company_id' in corecord._fields` is implied by being in `regular_fields`, and the second one does not make sense to me.

Reply:
It is because `check_company_domain` always returns a domain containing `company_id`; otherwise I have to manage the existence of the field in that function making it harder to override.